main.py

```
@@ -0,0 +1,94 @@
+                            sharepoint_file = value
+                            if sharepoint_file:
+                                session_id = f"sharepoint_download_{test_case['test']['name']}"
+                                post('preprocessor',
+                                     dict(rule="sharepoint_download", file_path=sharepoint_file, sid=session_id))
+                                state = get_host().get_state('preprocessor', session_id)
+                                if "exception" in state:
+                                    print(f"Error: {state}")
+                                    raise Exception(f"Error: {state['exception']}")
+
+            # Process the data if the required keys are present
+            source_df, target_df = process_data(test_case)
```

Ruff (0.8.2)

`process_data` may be undefined, or defined from star imports

(F405)